lib/util/errors.js

```diff
@@ -0,0 +1,186 @@
+
+  if (kuzzleError.class === 'PartialError') {
+    return new errors[kuzzleError.class](message, body, errorName, code);
+  }
+
+  return new errors[kuzzleError.class](message, errorName, code);
+}
+
+/**
+ * Construct and throw the corresponding error
+ *
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function throwError (domain, subdomain, error, ...placeholders) {
+  throw get(domain, subdomain, error, ...placeholders);
+}
+
+/**
+ * Returns a promise rejected with the corresponding error
+ *
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function reject (domain, subdomain, error, ...placeholders) {
+  return Bluebird.reject(get(domain, subdomain, error, ...placeholders));
+}
+
+/**
+ * Returns a promise rejected with the corresponding error, with its stack
+ * trace derivated from a provided source error
+ *
+ * @param  {Error}  source
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function rejectFrom (source, domain, subdomain, error, ...placeholders) {
+  return Bluebird.reject(
+    getFrom(source, domain, subdomain, error, ...placeholders));
+}
+
+/**
+ * Construct and return the corresponding error, with its stack
+ * trace derivated from a provided source error
+ *
+ * @param  {Error}  source
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function getFrom (source, domain, subdomain, error, ...placeholders) {
+  const derivedError = get(domain, subdomain, error, ...placeholders);
+
+  derivedError.stack = source.stack;
+
+  return derivedError;
+}
+
+/**
+ * Construct and throw the corresponding error, with its stack
+ * trace derivated from a provided source error
+ *
+ * @param  {string} source
+ * @param  {string} domain - Domain (eg: 'external')
+ * @param  {string} subdomain - Subdomain (eg: 'elasticsearch')
+ * @param  {string} error - Error name: (eg: 'index_not_found')
+ * @param  {...any} placeholders - Placeholders value to inject in error message
+ */
+function throwFrom (source, domain, subdomain, error, ...placeholders) {
+  const foo = getFrom(source, domain, subdomain, error, ...placeholders);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately throw this expression instead of assigning it to the temporary variable "foo". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)